pkg/fanal/analyzer/language/swift/swift/swift.go

```diff
@@ -0,0 +1,45 @@
+package swift
+
+import (
+	"context"
+	"os"
+
+	"github.com/aquasecurity/go-dep-parser/pkg/swift/swift"
+	"github.com/aquasecurity/trivy/pkg/fanal/analyzer"
+	"github.com/aquasecurity/trivy/pkg/fanal/analyzer/language"
+	"github.com/aquasecurity/trivy/pkg/fanal/types"
+
+	"golang.org/x/xerrors"
+)
+
+func init() {
+	analyzer.RegisterAnalyzer(&swiftLockAnalyzer{})
+}
+
+const (
+	version = 1
+)
+
+// swiftLockAnalyzer analyzes Package.resolved files
+type swiftLockAnalyzer struct{}
+
+func (a swiftLockAnalyzer) Analyze(_ context.Context, input analyzer.AnalysisInput) (*analyzer.AnalysisResult, error) {
+	p := swift.NewParser()
+	res, err := language.Analyze(types.Swift, input.FilePath, input.Content, p)
+	if err != nil {
+		return nil, xerrors.Errorf("%s parse error: %w", input.FilePath, err)
+	}
+	return res, nil
+}
+
+func (a swiftLockAnalyzer) Required(_ string, fileInfo os.FileInfo) bool {
+	return fileInfo.Name() == types.SwiftResolved
+}
```

Comment:
I think other analyzers use `filePath`.

```suggestion
func (a swiftLockAnalyzer) Required(filePath string, _ os.FileInfo) bool {
	return path.Base(filePath) == types.SwiftResolved
}
```

Reply:
hm.. i don't know why i used FileInfo...
Thanks! Changed in https://github.com/aquasecurity/trivy/pull/4932/commits/9d1e8b6bc23008767689c224d72faf30669af9f3